skfp/fingerprints/map.py

```diff
@@ -139,135 +144,124 @@ def __init__(
         X = ensure_mols(X)
         X = np.stack(
-            [self._calculate_single_mol_fingerprint(mol) for mol in X], dtype=int
+            [self._calculate_single_mol_fingerprint(mol) for mol in X],
+            dtype=np.uint32 if self.counts else np.uint8,
         )
 
-        if self.variant == "bit":
-            X = (X > 0).astype(np.uint8)
-        elif self.variant == "count":
-            X = X.astype(np.uint32)
-
         return csr_array(X) if self.sparse else np.array(X)
 
     def _calculate_single_mol_fingerprint(self, mol: Mol) -> np.ndarray:
         atoms_envs = self._get_atom_envs(mol)
-        shingles = self._get_atom_pair_shingles(mol, atoms_envs)
+        shinglings = self._get_atom_pair_shingles(mol, atoms_envs)
+
+        if self.counts:
+            folded = np.zeros(self.fp_size, dtype=np.uint32)
+            for shingling in shinglings:
+                hashed = struct.unpack("<I", sha256(shingling).digest()[:4])[0]
+                folded[hashed % self.fp_size] += 1
+            return folded
+
+        folded = np.zeros(self.fp_size, dtype=np.uint8)
+        for shingling in shinglings:
+            hashed = struct.unpack("<I", sha256(shingling).digest()[:4])[0]
+            folded[hashed % self.fp_size] = 1
+        return folded
+
+    @classmethod
+    def _find_env(cls, mol: Mol, atom_identifier: int, radius: int) -> Optional[str]:
+        """Returns a smile representation of the atom environment of a given radius.
+
+        Parameters
+        ----------
+        mol : Mol
+            The molecule to calculate the fingerprint for
+        atom_identifier : int
+            The index of the atom to calculate the environment for
+        radius : int
+            The radius of the environment
+
+        Returns
+        -------
+        str
+            The SMILES representation of the atom environment, or
+            None if the atom is not found
+        """
+        atom_identifiers_within_radius: list[int] = FindAtomEnvironmentOfRadiusN(
+            mol=mol, radius=radius, rootedAtAtom=atom_identifier
+        )
+        atom_map: dict = {}
 
-        if self.variant == "raw_hashes":
-            encoder = MinHash(num_perm=self.fp_size, seed=self.random_state)
-            encoder.update_batch(shingles)
-            fp = encoder.digest()
-        else:
-            # bit/count folded version from original MAP4 and MHFP implementation
-            hashes = [self._get_hash(shingle) for shingle in shingles]
-            bits = [hash_val % self.fp_size for hash_val in hashes]
-            fp = np.bincount(bits, minlength=self.fp_size)
+        sub_molecule: Mol = PathToSubmol(
+            mol, atom_identifiers_within_radius, atomMap=atom_map
+        )
+        if atom_identifier in atom_map:
+            smiles = MolToSmiles(
+                sub_molecule,
+                rootedAtAtom=atom_map[atom_identifier],
+                canonical=True,
+                isomericSmiles=False,
```

Comment:
Aren't those default values?

Reply:
Isn't it clearer if I leave it specified?

Reply:
On one hand - absolutely, it is. On the other, if you provide values, at least I immediately think "oh, so they are different than defaults", since we typically don't provide those, at least for common functions. So I would prefer to just use defaults

Reply:
Turns out that `isomericSmiles` by default is True, so I left the other kwarg. It was like this in the original map4 impl, I am not certain why it was set this way.

Reply:
Sure, just please leave a comment about this

Reply:
Done